src/account/accountsReducer.js

```diff
@@ -110,6 +112,33 @@ export default (state: AccountsState = initialState, action: Action): AccountsSt
     case ACCOUNT_REMOVE:
       return accountRemove(state, action);
 
+    case EVENT: {
+      const { event } = action;
+      switch (event.type) {
+        case 'restart': {
+          const { zulip_feature_level, zulip_version } = event;
+          return zulip_feature_level !== undefined && zulip_version !== undefined
+            ? [
+                {
+                  ...state[0],
+
+                  // TODO (?): Detect if these are different from the values
+                  // we had before, so we know it's an upgrade, not just a
+                  // restart. Then, implement logic like "be sure to refetch
+                  // from scratch within N hours" (to avoid thundering
+                  // herding the server).
```

Comment:
I think the "check if it's different" part is pretty valuable for load on servers that upgrade often from Git.

Also, I suspect we only need to check zulip_feature_level -- the Zulip version will change with every Git update.

Reply:
> I think the "check if it's different" part is pretty valuable for load on servers that upgrade often from Git.

That sounds right. This PR doesn't have us start doing a `/register` on restart events, it only updates the state, which we'll use for the rest of the session. Also, that state is persisted, so it'll inform the beginning of the next session, until that session naturally does its own `/register` which we'll grab a new value from.

So I think the load on servers is unchanged. When we do start having restart events cause a `/register`, we'll want to be careful about thundering herding.